src/posix/platform/netif.cpp

```diff
@@ -346,6 +351,8 @@ void AddRtAttr(struct nlmsghdr *aHeader, uint32_t aMaxLen, uint8_t aType, const
         memcpy(RTA_DATA(rta), aData, aLen);
     }
     aHeader->nlmsg_len = NLMSG_ALIGN(aHeader->nlmsg_len) + RTA_ALIGN(len);
+
+    return rta;
```

Comment:
I was wondering how do we return a pointer? Is it allocated somehow? Who owns it?

Looks like we cast the `aHeader`? 
```c++
rta  = (struct rtattr *)((char *)(aHeader) + NLMSG_ALIGN((aHeader)->nlmsg_len));
```

And `aHeader` always comes from `nh` in a:
```c++
    struct
    {
        struct nlmsghdr  nh;
        struct ifinfomsg ifi;
        char             buf[512];
    } req;
```

May make sense todefine this `struct` as a type and pass this instead? May be in a future PR?

Reply:
Yes, we may create a class to simplify generating netlink messages in a future PR.